### ISSUE TITLE
call-doc-and-style-r PR generates after code PR

### ISSUE THREAD

**Issue by efletcherPIFSC** [nmfs-ost/ghactions4r#30]
@k-doering-NOAA 

Our ss3diags repo is using the default call-doc-and-style-r action. The doc and r code styler action does not integrate with the code pull request, a separate pull request is generated for call-doc-and-style-r.

```
# document and style using reusable workflow
# This workflow runs devtools::document and styler::style_active_package(), then opens a pull request to the branch
# That started the workflow with the changes
name: call-doc-and-style-r
# on specifies the build triggers. See more info at https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows
on:
# workflow_dispatch requires pushing a button to run the workflow manually. uncomment the following line to add:
  #workflow_dispatch:
  # Runs the workflow on each push to the main or master branch:
  push:
    branches: [main, master]
jobs:
  call-workflow:
    uses: nmfs-fish-tools/ghactions4r/.github/workflows/doc-and-style-r.yml@main
```

Is there a setup were that the **Doc and R code styler** integrate with code puil requests (similar to R CMD CHECK)?

**Comment by k-doering-NOAA** [comment]
Thanks for bringing this up @efletcherPIFSC ! We could definitely add an option like that.

What do you think would be the ideal workflow? Is it that doc and style would run when pull requests are opened and the changes would be committed automatically to the branch? Or something else?

**Comment by k-doering-NOAA** [comment]
Thanks @efletcherPIFSC for the inspiration to extend call-doc-and-style-r. See #31.

To generate the workflow file needed that will commit directly on pull request, install the branch version of ghactions4r using remotes::install_github("nmfs-fish-tools/ghactions4r@test-commit-to-pr-doc-style", then run 
```r
ghactions4r::use_doc_and_style_r(how_to_commit = "directly", build_trigger = "pull_request")
```

If you want to also change $ references to [[]] (see ?rm_dollar_sign for more details), you can add the argument `use_rm_dollar_sign = TRUE` to the function call.

The branch name can be changed back to main once the pull request is merged in.

EDIT: these changes are now in the main branch, so removed text pertaining to when it was on a branch.

**Comment by efletcherPIFSC** [comment]
@k-doering-NOAA 

Apologies for the late reply. but the motivation to integrate the styler to pull requests was to lighten the pull request load,  using the "github workflow" (main-feature). ss3diags PR (and issue) numbers increased when the code styler was added. 

In addition, the change will prevent an issue cloning the "unstyled" repo before it was styled.

**Comment by k-doering-NOAA** [comment]
@efletcherPIFSC , that makes sense! I think the capabilities you need are now available in the doc and style workflow in ghactions4r. Let me know if you need additional guidance setting it up.

**Comment by efletcherPIFSC** [comment]
Successfully Applied docs and styler PR workflow to ss3diags. Note: Docs and styler adds a reviewable commit to the PR.
